recipe_scrapers/schoolofwok.py

```diff
@@ -0,0 +1,172 @@
+            .findChildren(
+                "div",
+                {
+                    "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                },
+            )[0]
+            .find_all("p")
+        )
+        ingredientslist = []
+
+        if possibleingredients:
+            for ingredient in possibleingredients:
+                if not ingredient.find("strong") and normalize_string(
+                    ingredient.get_text()
+                ):
+                    ingredientslist.append(normalize_string(ingredient.get_text()))
+        else:
+            possibleingredients = (
+                self.soup.find("section", {"id": "recipe-ingredients"})
+                .findChildren(
+                    "div",
+                    {
+                        "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                    },
+                )[0]
+                .find_all("li")
+            )
+            ingredientslist = [
+                normalize_string(ingredient.get_text())
+                for ingredient in possibleingredients
+            ]
+
+        return ingredientslist
+
+    def ingredient_groups(self):
+        possibleingredients = (
+            self.soup.find("section", {"id": "recipe-ingredients"})
+            .findChildren(
+                "div",
+                {
+                    "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                },
+            )[0]
+            .find_all("p")
+        )
+        groupings = defaultdict(list)
+        current_heading = None
+
+        if possibleingredients:
+            for ingredient in possibleingredients:
+                normalizedingredient = normalize_string(ingredient.get_text())
+                if normalizedingredient:
+                    if ingredient.find("strong"):
+                        current_heading = normalizedingredient
+                    else:
+                        groupings[current_heading].append(normalizedingredient)
+        else:
+            possibleingredients = (
+                self.soup.find("section", {"id": "recipe-ingredients"})
+                .findChildren(
+                    "div",
+                    {
+                        "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                    },
+                )[0]
+                .find_all()
+            )
+            for ingredient in possibleingredients:
+                normalizedingredient = normalize_string(ingredient.get_text())
+                if normalizedingredient:
+                    if (
+                        ingredient.name == "h3"
+                        and normalizedingredient.lower() != "ingredients"
+                    ):
+                        current_heading = normalizedingredient
+                    elif ingredient.name == "li":
+                        groupings[current_heading].append(normalizedingredient)
```

Comment:
```suggestion
        section = self.soup.find("section", {"id": "recipe-ingredients"})
        div_class = "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
        divs = section.findChildren("div", {"class": div_class})

        if not divs:
            return []

        paragraphs = divs[0].find_all("p")
        groupings = defaultdict(list)
        current_heading = None

        if paragraphs:
            for paragraph in paragraphs:
                normalized_ingredient = normalize_string(paragraph.get_text())
                if normalized_ingredient:
                    if paragraph.find("strong"):
                        current_heading = normalized_ingredient
                    else:
                        groupings[current_heading].append(normalized_ingredient)
        else:
            list_items = divs[0].find_all()
            for item in list_items:
                if (
                    item.name == "h3"
                    and normalize_string(item.get_text()).lower() != "ingredients"
                ):
                    current_heading = normalize_string(item.get_text())
                elif item.name == "li":
                    normalized_ingredient = normalize_string(item.get_text())
                    if normalized_ingredient:
                        groupings[current_heading].append(normalized_ingredient)

```
Same as ingredients, this is largely a readability refactor but it does also add specific handling for an instance where the div does not exist & consolidates a little duplicate code.